### PR TITLE
feat(framework:skip) Add function to fuse config overrides

### PR DESCRIPTION
**Pull request by charlesbvll**
### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: src/py/flwr/common/config.py [hunk 1]
```diff
@@ -73,6 +76,41 @@ def get_project_config(project_dir: Union[str, Path]) -> Dict[str, Any]:
     return config
 
 
+def get_fused_config(run: Run, flwr_dir: Optional[Path]) -> Dict[str, str]:
+    """Get the config using the fab_id and the fab_version, remove the nesting by adding
+    the nested keys as prefixes separated by dots, and fuse it with the override
+    dict."""
```
Comment (danieljanes) [inline src/py/flwr/common/config.py:82]: First line: Single-line single-sentence

File: src/py/flwr/common/config.py [hunk 2]
```diff
@@ -73,6 +76,43 @@ def get_project_config(project_dir: Union[str, Path]) -> Dict[str, Any]:
...
+    the nested keys as prefixes separated by dots, and fuse it with the override dict.
+    """
+    if not run.fab_id or not run.fab_version:
+        return {}
+
+    default_config = flatten_dict(
+        get_project_config(get_project_dir(run.fab_id, run.fab_version, flwr_dir))[
+            "flower"
+        ]["config"]
+    )
```
Comment (danieljanes) [inline src/py/flwr/common/config.py:92]: Let's split this on multiple lines to make it more readable

File: src/py/flwr/common/config.py [hunk 3]
```diff
@@ -73,6 +76,42 @@ def get_project_config(project_dir: Union[str, Path]) -> Dict[str, Any]:
...
+
+    Get the config using the fab_id and the fab_version, remove the nesting by adding
+    the nested keys as prefixes separated by dots, and fuse it with the override dict.
+    """
+    if not run.fab_id or not run.fab_version:
+        return {}
+
+    project_dir = get_project_dir(run.fab_id, run.fab_version, flwr_dir)
+
+    default_config = get_project_config(project_dir)["flower"]["config"]
```
Comment (danieljanes) [inline src/py/flwr/common/config.py:90]: A project might not have `config` set, we should check for that.

Comment (charlesbvll) [reply src/py/flwr/common/config.py:90]: Oh absolutely, sorry for that, I'll correct it

**Review by danieljanes — DISMISSED** [review]

**Review by danieljanes — APPROVED** [review]

**Pull request merged**